### PR TITLE
[Maintenance][API] Install API Platform 4.0

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
         "ext-intl": "*",
         "ext-json": "*",
         "ext-simplexml": "*",
-        "api-platform/core": "^3.4.2",
+        "api-platform/core": "^4.0",
```
Comment (GSadee) [inline composer.json]:
```suggestion
        "api-platform/core": "^4.0.3",
```
to be sure, that our fixes in XmlExtractor are available in APIP

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/admin/AdminUserPassword.xml [hunk 2]
```diff
@@ -48,11 +48,6 @@
                         </value>
                     </values>
                 </validationContext>
-                <openapiContext>
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/admin/AdminUserPassword.xml:51]: We need to cover these removed OpenApi contexts in different way, but we could add it separately 🖖🏻

File: src/Sylius/Bundle/ApiBundle/Resources/config/services.xml [hunk 3]
```diff
@@ -25,6 +25,9 @@
     </parameters>
 
     <services>
+        <service id="api_platform.path_segment_name_generator.underscore" class="ApiPlatform\Metadata\Operation\UnderscorePathSegmentNameGenerator" />
+        <service id="api_platform.path_segment_name_generator.dash" class="ApiPlatform\Metadata\Operation\DashPathSegmentNameGenerator" />
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services.xml]: Are these services needed? 🤔

**Review by GSadee — DISMISSED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/admin/AdminUserPassword.xml (hunk 2 above)
Comment (mpysiak) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/admin/AdminUserPassword.xml:51]: OpenapiContext is not available in the schema in 4.0

**Review by GSadee — APPROVED** [review]

**Review by NoResponseMate — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/admin/AdminUserPassword.xml (hunk 2 above)
Comment (Rafikooo) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/admin/AdminUserPassword.xml:51]: The `openapiContext` is now accessible under the name `openapi`:
https://github.com/api-platform/core/blob/3.4/src/OpenApi/Factory/OpenApiFactory.php#L233

**Pull request merged**